functions/Export-DbaDacpac.ps1

```diff
@@ -0,0 +1,179 @@
+
+	Sets the CommandTimeout to 10 then extracts the dacpac for SharePoint_Config on sql2016 to C:\temp\SharePoint_Config.dacpac then verifies extraction.
+
+	
+    #>
+	[CmdletBinding()]
+	param
+	(
+		[parameter(Mandatory, ValueFromPipeline)]
+		[Alias("ServerInstance", "SqlServer")]
+		[DbaInstanceParameter[]]$SqlInstance,
+		[Alias("Credential")]
+		[PSCredential]$SqlCredential,
+		[object[]]$Database,
+		[object[]]$ExcludeDatabase,
+		[switch]$AllUserDatabases,
+		[string]$Directory = "$home\Documents",
+		[string]$ExtendedParameters,
+		[string]$ExtendedProperties,
+		[switch]$EnableException
+	)
+	
+	process {
+		if (!$Database -and !$ExcludeDatabase -and !$AllUserDatabases) {
+			Stop-Function -Message "You must specify databases to execute against using either -Database, -ExcludeDatabase or -AllUserDatabases" -Continue
+		}
+		
+		if (-not (Test-Path $Directory)) {
+			Stop-Function -Message "$Directory doesn't exist or access denied" -Continue
+		}
+		
+		foreach ($instance in $sqlinstance) {
+			
+			try {
+				Write-Message -Level Verbose -Message "Connecting to $instance."
+				$server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $sqlcredential -MinimumVersion 9
+			}
+			catch {
+				Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+			}
+			
+			$dbs = $server.Databases | Where-Object { $_.IsSystemObject -eq $false }
+			
+			if ($Database) {
+				$dbs = $dbs | Where-Object Name -in $Database
+			}
+			
+			if ($ExcludeDatabase) {
+				$dbs = $dbs | Where-Object Name -notin $ExcludeDatabase
+			}
+			
+			foreach ($db in $dbs) {
+				$dbname = $db.name
+				$connstring = $server.ConnectionContext.ConnectionString.Replace('"', "'")
+				if ($connstring -notmatch 'Database=') {
+					$connstring = "$connstring;Database=$dbname"
+				}
+				$path = "$Directory\$dbname.dacpac"
+				Write-Message -Level Verbose -Message "Exporting $path"
+				Write-Message -Level Verbose -Message "Using connection string $connstring"
+				
+				$sqlPackageArgs = "/action:Extract /tf:""$path"" /SourceConnectionString:""$connstring"" $ExtendedParameters $ExtendedProperties"
+				
+				try {
+					$resultstime = [diagnostics.stopwatch]::StartNew()
+					
+					$startprocess = New-Object System.Diagnostics.ProcessStartInfo
+					$startprocess.FileName = "$script:PSModuleRoot\bin\dacfx\sqlpackage.exe"
+					$startprocess.Arguments = $sqlPackageArgs
+					$startprocess.RedirectStandardError = $true
+					$startprocess.RedirectStandardOutput = $true
+					$startprocess.UseShellExecute = $false
+					$process = New-Object System.Diagnostics.Process
+					$process.StartInfo = $startprocess
+					$process.Start() | Out-Null
+					$process.WaitForExit()
+					$stdout = $process.StandardOutput.ReadToEnd()
+					$stderr = $process.StandardError.ReadToEnd()
+					Write-Message -level Verbose -Message "StandardOutput: $stdout"
+					
+					[pscustomobject]@{
```

Comment:
It would be cleaner to just set the `$status` in the `try{} catch{}` and define the `PSCustomObject` and DefaultView only once after that block.

Reply:
thank you, i fixed the $elapsed but the Stop-Function makes it so that there wouldnt be an output of the object if it failed. Which is alright if that's the way we wanna go.